crates/rooch-types/src/transaction/rooch.rs

```diff
@@ -86,10 +87,77 @@ impl Display for RoochTransactionData {
     }
 }
 
+/// PartiallySignedRoochTransaction(PSRT) is a transaction that has been signed by partial signers.
+/// It can be used for multi-signatures.
+#[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+pub struct PartiallySignedRoochTransaction {
+    pub data: RoochTransactionData,
+    /// The threshold of the signatures.
+    pub threshold: u64,
+    /// The signatures of the partial signers.
+    pub authenticators: Vec<BitcoinAuthenticator>,
+}
+
+impl PartiallySignedRoochTransaction {
+    pub fn new(data: RoochTransactionData, threshold: u64) -> Self {
+        Self {
+            data,
+            threshold,
+            authenticators: vec![],
+        }
+    }
+
+    pub fn sender(&self) -> RoochAddress {
+        self.data.sender
+    }
+
+    pub fn contains_authenticator(&self, authenticator: &BitcoinAuthenticator) -> bool {
+        self.authenticators
+            .iter()
+            .any(|a| a.payload.public_key == authenticator.payload.public_key)
+    }
+
+    pub fn add_authenticator(&mut self, authenticator: BitcoinAuthenticator) -> Result<()> {
```

Comment:
Is there a way to remove the authenticator if not needed?

Reply:
No usecase